homeassistant/components/shelly/__init__.py

```diff
@@ -479,10 +495,41 @@ def __init__(
         self.entry = entry
         self.device = device
 
+        self._async_remove_device_updates_handler = self.async_add_listener(
```

Comment:
When is this called? And why not add it to `entry.async_on_unload` ?

Reply:
Fixed